### ISSUE TITLE
Notefox accesses Google servers to download Google fonts

### ISSUE THREAD

**Issue by Cade66** [Sav22999/websites-notes#67]
Hi,

I've just realized when looking at my connection logs that Notefox connects to fonts.googleapis.com retrieve fonts.
Personally *googleapis* is blocked system-wide here (as most Google servers).
Sites which require Google fonts are handled by the LocalCDN extension (Firefox) but of course the extension does not handle call to Google servers initiated by other extensions... such as Notefox.
Given I block connections to fonts.googleapis.com, every time I open Notefox (be it the popup only) I get four connection attempts to dedicated Google server.
I run 60+ extensions and this is the first time I notice an extension proceeding this way.
IMO all fonts are to be included in the very extension, calling a server for whatever is not as I see it a correct way of proceeding.

**Comment by Sav22999** [comment]
@Cade66 Before all fonts where included. You think if I use a different CDN is better or is better include directly in the addon?

**Comment by Cade66** [comment]
@Sav22999 if you ask me, my opinion is that an extension should always directly include its fonts. And not only its fonts : unless required for the very functionality of the add-on (such as data update i.e. IP databases) an extension should NOT access servers for whatever reason, and especially not Google servers. More specifically concerning fonts users may wish to keep/use one font only (for the browser, for the extensions, sometimes as well for all sites. Some sites need specific fonts and that's where an add-on such as LocalCDN comes in handy (and not only for fonts). We all know that connections to 3rd-party servers is tremendous so limiting this (LocalCDN, uBO) is IMO a fundamental principle.

**Comment by Sav22999** [comment]
In the next release I'll remove again every Google CDN (fonts) 😄 
Thanks!

**Comment by Sav22999** [comment]
@Cade66 About the use of many fonts... actually my addon use only one font: Open Sans, but it has many "styles" (for bold, for italic, etc.) and I use "Source Code Pro" (which is open-source too) for "monospace" family -> (I'm trying "Victor Mono" recently as monospace, which is also open-source, but I actually prefer the first one).

**Comment by Cade66** [comment]
@Sav22999 , several fonts, one font with many srtyles ... anyway I was sharing my thoughts in a general scope of extensions not specifically regarding Notefox. Again I emphasize that i'm aware this approach of mine is, as often when it comes to an extension's style, concerns mainly cosmetics and therefor triggers subjectivity. Anyway that was a side-note, not an argument, useless in fact. My main concern was fonts' download, and i'm happy that next release will include its fonts, natively so to say.

One word about latest Notefox ver. 3.7.11 : i'm running 3.7.10, when updating from the browser (about:addons) 3.7.11 was notified as available, I updated but the update turned out to be, to remain 3.7.10 (likely a Mozilla mistake)  ... I went on AMO, downloaded 3.7.11 and installed it manually. That's when odd things started : high CPU, lasting. I closed Firefox, reopened it, again high CPU. That's when I had a look at my connections and discovered blocked connections to fonts.googleapis.com. I thought about Notefox 3.7.11 just updated. I closed Firefox, restored my profile (I always backup profile before any extension update) and restarted Firefox hence with 3.7.10 : no more high CPU. No idea what in Notefox 3.7.11 caused this, not even sure its related to fonts download given previous versions of Notefox did (try to) download fonts as well without this CPU enigma. This is FYI of course. Maybe a new way of connecting to Google servers? No idea, i'd have to dig into the code and I'm at this time totally elsewhere so to say :)
